vault/expiration_test.go

```diff
@@ -2491,3 +2491,165 @@ func TestExpiration_FairsharingEnvVar(t *testing.T) {
 		}
 	}
 }
+
+// register one lease ID and return the leaseID
+func registerOneLease(t *testing.T, ctx context.Context, exp *ExpirationManager) string {
```

Comment:
The general question I have, looking at these tests, is that I'm not sure why we are tracking zombies separately in a separate struct. From what I can tell, we're still treating them in a similar way, modulo a few operations where we aren't doing that operation on the zombie lease (because it's stored separately). Is that understanding correct? If so, then wouldn't it be more clear to keep zombies as pending leases, but check whether a lease is a zombie using isZombie(), and skip actions for that lease if it is? 
Thanks!

Reply:
I think the key difference is that zombies are a separate class of lease. On Vault start we load leases in from storage, and handle zombies completely differently than non-zombies. I think it makes it a little bit more clear and less prone to bugs to not re-use the pending map (since pending is so deeply integrated). The zombies also won't have a timer to trigger revocation attempts.

Basically the expiration manager is actively managing the pending leases, but is doing basically nothing with the zombie leases (besides (in the future) exposing them via an API call to be force revoked).

Reply:
to some extent you can think of marking a lease as a zombie as being similar to tomb-stoning in a database (except that zombies must be manually deleted). we want to get it out of rotation and there's no reason to expose it with all of the other valid leases in rotation